topostats/scars.py

```diff
@@ -0,0 +1,445 @@
+        that result in values in the mask higher than this threshold are automatically considered scars.
+    max_scar_width: int
+        An integer that restricts the algorithm to only mark scars that are as thin or thinner than this width.
+        This is important for ensuring that legitimate grain data is not detected as scarred data.
+        Note that width here is vertical, as scars are thin, horizontal features.
+    min_scar_length: int
+        An integer that restricts the algorithm to only mark scars that are as long or longer than this length.
+        This is important for ensuring that noise or legitimate but sharp datapoints do not get detected as scars.
+        Note that length here is horizontal, as scars are thin, horizontal features.
+    Returns
+    -------
+    marked: np.ndarray
+        Returns a 2-D image of the same shape as the data image, where each pixel's value represents
+        a metric for how strongly that pixel is considered to be a scar. Higher values mean more likely
+        to be a scar.
+    """
+
+    image = np.copy(img)
+
+    stddev = np.std(image)
+    marked = np.zeros(image.shape)
+
+    for row in range(image.shape[0] - 1):
+        for col in range(image.shape[1]):
+            if direction == "positive":
+                _mark_if_positive_scar(
+                    row_col=(row, col),
+                    stddev=stddev,
+                    img=image,
+                    marked=marked,
+                    threshold_low=threshold_low,
+                    max_scar_width=max_scar_width,
+                )
+
+            elif direction == "negative":
+                _mark_if_negative_scar(
+                    row_col=(row, col),
+                    stddev=stddev,
+                    img=image,
+                    marked=marked,
+                    threshold_low=threshold_low,
+                    max_scar_width=max_scar_width,
+                )
+
+            else:
+                raise ValueError(f"direction {direction} invalid.")
+
+    _spread_scars(marked=marked, threshold_low=threshold_low, threshold_high=threshold_high)
+
+    _remove_short_scars(marked=marked, threshold_high=threshold_high, min_scar_length=min_scar_length)
+
+    return marked
+
+
+def _remove_marked_scars(img: np.ndarray, scar_mask: np.ndarray) -> None:
+    """
+    Interpolate values covered by marked scars. Takes an image, and a marked scar boolean mask for that
+    image. Returns the image where the marked scars are replaced by interpolated values.
+
+    Parameters
+    ----------
+    img: np.ndarray
+        A 2-D image of the data to remove scars from.
+    scar_mask: np.ndarray
+        A boolean image of pixels that determine which values are flagged as scars and therefore should
+        be interpolated over in the original data image.
+
+    Returns
+    -------
+    None
+    """
+
+    for row in range(img.shape[0]):
```

Comment:
Could iterate directly over arrays with...
```suggestion
for row, col in np.index(img.shape):
```